backends-velox/src/test/scala/org/apache/gluten/execution/ScalarFunctionsValidateSuite.scala

```diff
@@ -1077,4 +1077,25 @@ class ScalarFunctionsValidateSuite extends FunctionsValidateTest {
         }
     }
   }
+
+  test("PreciseTimestampConversion") {
+    withTempPath {
+      path =>
+        val df = spark
+          .sql(
+            "select * from VALUES ('A1', TIMESTAMP'2021-01-01 00:00:00'), " +
+              "('A1', TIMESTAMP'2021-01-01 00:04:30'), ('A1', TIMESTAMP'2021-01-01 00:06:00'), " +
+              "('A2', TIMESTAMP'2021-01-01 00:01:00') AS tab(a, b)")
+          .write
+          .parquet(path.getCanonicalPath)
+
+        spark.read.parquet(path.getCanonicalPath).createOrReplaceTempView("string_timestamp")
+
+        runQueryAndCompare(
+          "SELECT a, window.start, window.end, count(*) as cnt FROM" +
+            " string_timestamp GROUP by a, window(b, '5 minutes') ORDER BY a, start;") {
+          checkGlutenOperatorMatch[ProjectExecTransformer]
```

Comment:
Is the round-trip conversion tested in this test?

Reply:
Yes, below is the fallback before this change.
```
(9) Project [codegen id : 2]
Output [2]: [named_struct(start, knownnullable(precisetimestampconversion(((precisetimestampconversion(b#28, TimestampType, LongType) - CASE WHEN (((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) < 0) THEN (((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) + 300000000) ELSE ((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) END) - 0), LongType, TimestampType)), end, knownnullable(precisetimestampconversion((((precisetimestampconversion(b#28, TimestampType, LongType) - CASE WHEN (((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) < 0) THEN (((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) + 300000000) ELSE ((precisetimestampconversion(b#28, TimestampType, LongType) - 0) % 300000000) END) - 0) + 300000000), LongType, TimestampType))) AS window#33, a#27]
Input [2]: [a#27, b#28]
```